api.py

```diff
@@ -59,10 +76,40 @@ def get(self, ipv4: str):
 
         return result
 
+
+    def _get_geoip2_service(self, ipv4: str):
+        client = geoip2.webservice.Client(
+            GEOIP2_USER_ID,
+            GEOIP2_LICENSE_KEY,
+        )
+        try:
+            response = client.insights(ipv4)
+            result['ip'] = ipv4
+            result['region_code'] =  self._text_escaper(response.subdivisions[0].iso_code)
+            result['region_name'] = self._text_escaper(response.subdivisions[0].names['en'])
+            result['city'] = self._text_escaper(response.city.names['en'])
+            result['location']['geonameid'] = self._text_escaper(response.city.names['en'])
+            result['zip'] = self._text_escaper(response.postal.code)
+            result['continent_code'] = self._text_escaper(response.continent.code)
+            result['continent_name'] = self._text_escaper(response.continent.names['en'])
+            result['country_code'] = self._text_escaper(response.country.iso_code)
+            result['country_name'] = self._text_escaper(response.country.names['en'])
+            result['latitude'] = response.location.latitude
+            result['longitude'] = response.location.longitude
+            result['location']['time_zone'] = self._text_escaper(response.location.time_zone)
+
+        except GeoIP2Error as error:
+            logging.error(error)
+            raise falcon.HTTPBadRequest('message', 'Invalid IPv4 address')
+        return result
+
+
+    def _text_escaper(self, text: None):
+        return html.escape(text, quote=True) if text else ''
```

Comment:
why do you need to escape text here?
Clients do not always need this and clients can escape text if they want to.

Reply:
It will break when there is a quote in country name, this code is also deployed for both tc, and retail

Reply:
the old code didn't escape text. How was it fine?

Reply:
NON paid version GEOIP dont have quotes.. while GEOIP2 paid version have a lot more details that contains this quotes

Reply:
ok, it's fine to do this for now